SwiftStyleGuide.swift

```diff
@@ -0,0 +1,83 @@
+//
+//  SwiftStyleGuide.swift
+//  PurchasesCoreSwift
+//
+//  Created by Andrés Boedo on 7/12/21.
+//  Copyright © 2021 Purchases. All rights reserved.
+//
+
+import Foundation
+
+protocol ValuePrintable {
+    // make protocol variables read-only unless needed
+    func foo() -> String
+}
+
+// use protocol extensions for default implementations of protocols for types
+extension ValuePrintable where Self: NSNumber {
+    func foo() -> String {
+        return "the number as float is: \(self.floatValue)"
+    }
+}
+
+// prefer structs to classes whenever possible.
+// keep in mind that structs are value types.
+// More info here: https://developer.apple.com/documentation/swift/choosing_between_structures_and_classes
+struct MyStruct {
+
+    // don't explicitly define types unless needed.
+    // prefer let to var whenever possible.
+
+    // public properties, then internal, then private.
+    public let mercury = false
+    // use `maybe` prefix for optionals
+    public var maybeVenus: String?
```

Comment:
I never did this, but I like it.

Reply:
I saw it on some apple videos and thought it was a good idea. But I think the context for those was local variables, I don't think I've ever seen it on a property, happy to update

Reply:
I'm a fan, let's do keep it 😄